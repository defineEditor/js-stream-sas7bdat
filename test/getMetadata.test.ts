import DatasetSas7BDat from '../src/index';
import path from 'path';

test('Get metadata', async () => {
    const filePath = path.join(__dirname, '/data/sample.sas7bdat');

    const data = new DatasetSas7BDat(filePath);
    const metadata = await data.getMetadata();
    expect(metadata).toMatchSnapshot();
});