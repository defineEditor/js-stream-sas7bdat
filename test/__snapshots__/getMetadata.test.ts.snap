// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get metadata 1`] = `
{
  "columns": [
    {
      "dataType": "string",
      "itemOID": "IT.Name",
      "label": "Name",
      "length": 8,
      "name": "Name",
    },
    {
      "dataType": "string",
      "itemOID": "IT.Sex",
      "label": "Sex",
      "length": 1,
      "name": "Sex",
    },
    {
      "dataType": "double",
      "displayFormat": "F4",
      "itemOID": "IT.Age",
      "label": "Age",
      "length": 8,
      "name": "Age",
    },
    {
      "dataType": "double",
      "displayFormat": "F8",
      "itemOID": "IT.Height",
      "label": "Height",
      "length": 8,
      "name": "Height",
    },
    {
      "dataType": "double",
      "itemOID": "IT.Weight",
      "label": "Weight",
      "length": 8,
      "name": "Weight",
    },
  ],
  "datasetJSONCreationDateTime": "2019-09-09T13:43:00.000Z",
  "datasetJSONVersion": "",
  "dbLastModifiedDateTime": "2019-09-12T11:15:54.000Z",
  "label": "CLASS",
  "name": "sample",
  "records": 19,
}
`;
